src/main/kotlin/io/github/mojira/arisa/modules/privacy/PrivacyModule.kt

```diff
@@ -0,0 +1,242 @@
+    }
+
+    private fun Issue.hasAnyAttachmentName(name: String) = attachments.any { it.name == name }
+
+    private fun redactAttachments(issue: Issue, attachments: Collection<RedactedAttachment>): Boolean {
+        var redactedAll = true
+        attachments
+            // Group by uploader in case they uploaded multiple attachments at once
+            .groupBy { it.attachment.uploader?.name!! }
+            .forEach { (uploader, userAttachments) ->
+                val fileNames = mutableSetOf<String>()
+                userAttachments.forEach {
+                    val attachment = it.attachment
+                    val tempDir = Files.createTempDirectory("arisa-redaction-upload").toFile()
+                    val fileName = "redacted_${attachment.name}"
+                    val filePath = getSafeChildPath(tempDir, fileName)
+
+                    if (filePath == null || issue.hasAnyAttachmentName(fileName) || !fileNames.add(fileName)) {
+                        redactedAll = false
+                        // Note: Don't log file name to avoid log injection
+                        log.warn("Cannot redact attachment with ID ${attachment.id} of issue ${issue.key}; file name " +
+                            "is malformed or would clash with other attachment")
+                        tempDir.delete()
+                    } else {
+                        log.info("Redacting attachment with ID ${attachment.id} of issue ${issue.key} because it " +
+                            "contains sensitive data")
+                        filePath.writeText(it.redactedContent)
+                        issue.addAttachmentFromFile(filePath) {
+                            // Once uploaded, delete the temp directory containing the attachment
+                            tempDir.deleteRecursively()
+                        }
+                        // Remove the original attachment
+                        attachment.remove()
+                    }
+                }
+
+                if (fileNames.isNotEmpty()) {
+                    // Use postfix line break to prevent bot signature from appearing as part of last list item
+                    val fileNamesString = fileNames.joinToString(separator = "", postfix = "\n") {
+                        // Use link for attachments
+                        "\n- [^${sanitizeCommentArg(it)}]"
+                    }
+                    val sanitizedUploaderName = sanitizeCommentArg(uploader)
+                    // Does not use helper message because message will only be used by bot and helper messages
+                    // currently only support one placeholder
+                    issue.addRawBotComment(
+                        "@[~$sanitizedUploaderName], sensitive data has been removed from your " +
+                            "attachment(s) and they have been re-uploaded as:$fileNamesString"
+                    )
```

Comment:
Message is hardcoded instead of using helper message because:
- It uses two placeholders but helper messages only support one at the moment
- This message is only useful for Arisa, as helper message it does not add that much value since the user would have to format the `$fileNamesString` filltext manually.